core/src/main/java/org/elasticsearch/index/indexing/ShardIndexingService.java

```diff
@@ -124,19 +128,31 @@ public void postCreate(Engine.Create create) {
             try {
                 listener.postCreate(create);
             } catch (Exception e) {
-                logger.warn("post listener [{}] failed", e, listener);
+                logger.warn("postCreate listener [{}] failed", e, listener);
             }
         }
     }
 
     public void postCreate(Engine.Create create, Throwable ex) {
+        for (IndexingOperationListener listener : listeners) {
+            try {
+                listener.postCreate(create, ex);
+            } catch (Exception e) {
+                logger.warn("postCreate listener [{}] failed", e, listener);
+            }
+        }
     }
 
     public Engine.Index preIndex(Engine.Index index) {
         totalStats.indexCurrent.inc();
         typeStats(index.type()).indexCurrent.inc();
         for (IndexingOperationListener listener : listeners) {
-            index = listener.preIndex(index);
+            try {
+                listener.preIndex(index);
+            } catch (Exception e) {
```

Comment:
@s1monw I think in preIndex we should not remove "try" in listeners. For example: I create percolator "search request". And this request fail - next fail "index request" it is bad.
I think we can make strict  for postIndex, postDelete, postCreate and preCreate.